cdap-ui/app/cdap/components/FastAction/ExploreAction/ExploreModal.js

```diff
@@ -209,13 +196,15 @@ export default class ExploreModal extends Component {
           <td> {query.statement}</td>
           <td>
             <div className="btn-group">
-              <button
-                className="btn btn-default"
-                onClick={this.downloadQuery.bind(this, query)}
-                disabled={!query.is_active || query.status !== 'FINISHED' ? 'disabled' : null}
-              >
-                <i className="fa fa-download"></i>
-              </button>
+              <a href={this.getDownloadUrl(query)}>
+                <button
+                  className="btn btn-default"
+                  onClick={this.fetchAndUpdateQueries.bind(this)}
+                  disabled={!query.is_active || query.status !== 'FINISHED' ? 'disabled' : null}
```

Comment:
you have to disable the button on the anchor level. Or don't use the button at all actually